### PR TITLE
Improve error handling in integration tests, update deps

### PR DESCRIPTION
**Pull request by fearful-symmetry**
This does two things:

- update elastic-agent-libs dependency to bring in https://github.com/elastic/elastic-agent-libs/pull/129
- Update the code that uses the fleet APIs where needed
- Improve error handling, mostly to make debugging easier while I deal with further bugs

## Why is it important?

We want to bring in the kibana API fixes, and make error handling easier.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

### CODE REVIEW

**Comment by fearful-symmetry** [comment]
Can't reproduce the CI errors locally, `make update` doesn't change anything.

File: go.mod [hunk 1]
```diff
@@ -12,13 +12,15 @@ require (
 	github.com/coreos/go-systemd/v22 v22.3.3-0.20220203105225-a9a7ef127534
 	github.com/docker/go-units v0.5.0
 	github.com/dolmen-go/contextio v0.0.0-20200217195037-68fc5150bcd5
+	github.com/elastic/beats/v7 v7.17.10
 	github.com/elastic/e2e-testing v1.99.2-0.20221205111528-ade3c840d0c0
 	github.com/elastic/elastic-agent-autodiscover v0.6.1
 	github.com/elastic/elastic-agent-client/v7 v7.1.2
-	github.com/elastic/elastic-agent-libs v0.3.8-0.20230512004837-d6a37d929cd6
+	github.com/elastic/elastic-agent-libs v0.3.9-0.20230606211614-a752b18efc9e
```
Comment (ycombinator) [inline go.mod:19]: Now that there's a `0.3.9` release, I think you can just use that here.
```suggestion
	github.com/elastic/elastic-agent-libs v0.3.9
```

File: changelog/fragments/1686161490-test-logging-dep-update.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+# PR URL; optional; the PR number that added the changeset.
+# If not present is automatically filled by the tooling finding the PR where this changelog fragment has been added.
+# NOTE: the tooling supports backports, so it's able to fill the original PR number instead of the backport PR number.
+# Please provide it if you are adding a fragment for a different PR.
+pr: https://github.com/elastic/elastic-agent/pull/2803
+
+# Issue URL; optional; the GitHub issue related to this changeset (either closes or is part of).
+# If not present is automatically filled by the tooling with the issue linked to the PR number.
+#issue: https://github.com/owner/repo/1234
```
Comment (ycombinator) [inline changelog/fragments/1686161490-test-logging-dep-update.yaml:32]: I think we should omit this CHANGELOG fragment as the changes here are not user-facing.  If you do this, you'll want to add the `skip-changelog` label to the PR to make CI happy.  WDYT?

Comment (fearful-symmetry) [reply changelog/fragments/1686161490-test-logging-dep-update.yaml:32]: Ahhh, okay. Was just trying to make CI happy, hah

**Review by ycombinator — APPROVED** [review]

**Comment by ycombinator** [comment]
We are looking to keep 8.8 in-sync with CI integration testing, so I have added the backport label for 8.8.

**Pull request merged**